### PR TITLE
Make test_mxnet.py fit for MXNet 2.0

### LINKED ISSUE #3061: Make these tests work with MXNet 2.0
The following tests do not work with MXNet 2.0, so they had to be disabled for `mxnet>=2.0`:

https://github.com/horovod/horovod/pull/3062/files#diff-860f0f56a942caffffe210a12419165434283275164e08b6cab3c25848e0c391

- test/parallel/test_mxnet.test_horovod_broadcast_deferred_init_parameters
- test/parallel/test_mxnet.test_two_trainer
- test/parallel/test_mxnet.test_gluon_trainer

Please fix and enable them again.

### PR DESCRIPTION
**Pull request by barry-jin**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes #3061.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by EnricoMi** [comment]
@barry-jin thanks for looking into this. These tests run against various MXNet version, all the way back to 1.5.1.

So either the tests are written in a way they work with all versions, or those modified tests are cloned to a mxnet 2.0 version which runs mutually exclusive (skipIf).

**Comment by barry-jin** [comment]
> @barry-jin thanks for looking into this. These tests run against various MXNet version, all the way back to 1.5.1.
> 
> So either the tests are written in a way they work with all versions, or those modified tests are cloned to a mxnet 2.0 version which runs mutually exclusive (skipIf).

@EnricoMi I see. Because there are a lot of breaking changes in MXNet2.0, especially for Gluon API and array objects. I'm wondering if we can create a new test file (test_mxnet2.py) for MXNet2.0 tests.

File: test/parallel/test_mxnet2.py [hunk 1]
```diff
@@ -0,0 +1,1546 @@
...
+    has_gpu = False
+    HAS_MXNET = False
+
+# Set environment variable to enable adding/removing process sets after initializing Horovod.
+os.environ["HOROVOD_DYNAMIC_PROCESS_SETS"] = "1"
+
+
+@pytest.mark.skipif(not HAS_MXNET, reason='MXNet unavailable')
+@pytest.mark.skipif(LooseVersion(mx.__version__) < LooseVersion('2.0.0'), reason='MXNet v2 tests')
+class MXTests(unittest.TestCase):
```
Comment (EnricoMi) [inline test/parallel/test_mxnet2.py:51]: Can we call this `MX2Tests` and derive it from `MXTests` so we don't copy all that code and have to maintain it in two places? Then `MX2Tests` can override and reimplement those tests that don't work in MXNet 2.0 only.

Will be interesting how the `@pytest.mark.skipif` handles the class hierarchy.

**Review by EnricoMi — CHANGES_REQUESTED** [review]

**Comment by EnricoMi** [comment]
@barry-jin thanks for the MXNet 2 fixes. I have reworked the `test_mxnet.py` in #3071, which includes your contribution. We can finish this fix there or you merge it into your branch and we finish it here. Either way, your contribution should be acknowledge when we merge my branch into master.

**Comment by barry-jin** [comment]
@EnricoMi Thanks! Let's finish the fix in #3072

**Pull request closed without merging**